gpflux/layers/basis_functions/fourier_features/random/base.py

```diff
@@ -25,17 +23,54 @@
 
 from gpflux.layers.basis_functions.fourier_features.base import FourierFeaturesBase
 from gpflux.layers.basis_functions.fourier_features.utils import (
-    ORF_SUPPORTED_KERNELS,
-    RFF_SUPPORTED_KERNELS,
     _bases_concat,
     _bases_cosine,
-    _ceil_divide,
     _matern_number,
-    _sample_chi,
-    _sample_students_t,
 )
 from gpflux.types import ShapeType
 
+"""
+Kernels supported by :class:`RandomFourierFeatures`.
+
+You can build RFF for shift-invariant stationary kernels from which you can
+sample frequencies from their power spectrum, following Bochner's theorem.
+"""
+RFF_SUPPORTED_KERNELS: Tuple[Type[gpflow.kernels.Stationary], ...] = (
+    gpflow.kernels.SquaredExponential,
+    gpflow.kernels.Matern12,
+    gpflow.kernels.Matern32,
+    gpflow.kernels.Matern52,
+)
+
+
+def _sample_students_t(nu: float, shape: ShapeType, dtype: DType) -> TensorType:
+    """
+    Draw samples from a (central) Student's t-distribution using the following:
+      BETA ~ Gamma(nu/2, nu/2) (shape-rate parameterization)
+      X ~ Normal(0, 1/BETA)
+    then:
+      X ~ StudentsT(nu)
+
+    Note this is equivalent to the more commonly used parameterization
+      Z ~ Chi2(nu) = Gamma(nu/2, 1/2)
+      EPSILON ~ Normal(0, 1)
+      X = EPSILON * sqrt(nu/Z)
+
+    To see this, note
+      Z/nu ~ Gamma(nu/2, nu/2)
+    and
+      X ~ Normal(0, nu/Z)
+    The equivalence becomes obvious when we set BETA = Z/nu
```

Comment:
It would be useful to add return shapes and docstrings for at least `nu`.

Reply:
I think I will address this in another improvement PR. I want to constrain this PR to the basic restructuring of modules.